src/contexts/VideoContext.jsx

```diff
@@ -1,18 +1,47 @@
-import { createContext, useContext, useReducer } from "react";
-import { videoReducer } from "../reducers/videoReducer";
+import axios from "axios";
+import {
+  createContext,
+  useContext,
+  useReducer,
+  useEffect,
+  useState,
+} from "react";
+import { videoReducer } from "reducers/videoReducer";
 import {
   addToWatchlist,
   removeFromWatchlist,
-} from "../router/utils/HelperFunctions";
+  addToLikedList,
+  removeFromLikedList,
+} from "router/utils/HelperFunctions";
 
 const VideoContext = createContext();
+
 const initialData = {
   watchList: [],
   likedList: [],
   history: [],
 };
+
 const VideoProvider = ({ children }) => {
   const [videoData, videoDispatch] = useReducer(videoReducer, initialData);
+  const [videoList, setVideoList] = useState([]);
+  const [showOptions, setShowOptions] = useState(false);
+
+  useEffect(() => {
+    const fetchVideos = async () => {
```

Comment:
you could shift all the API calls to a different files will make the code modular